src/explore-education-statistics-common/src/modules/charts/components/__tests__/MapBlockInternal.test.tsx

```diff
@@ -334,28 +318,89 @@ describe('MapBlock', () => {
         draft.data = testFullTableRegion.results;
       });
 
-      render(<MapBlock {...testBlockPropsRegion} />);
+      render(<MapBlockInternal {...testBlockPropsRegion} />);
 
       await waitFor(() => {
         expect(screen.getByLabelText('2. Select a Region')).toBeInTheDocument();
       });
     });
 
     test('shows the default label if the data set contains multiple types', async () => {
-      const testFullTableRegion = mapFullTable(testMapTableDataMixed);
+      const testFullTableMixed = mapFullTable(testMapTableDataMixed);
+      const testBlockPropsMixed = produce(testBlockProps, draft => {
+        draft.meta = testFullTableMixed.subjectMeta;
+        draft.data = testFullTableMixed.results;
+      });
 
-      const testBlockPropsRegion = produce(testBlockProps, draft => {
-        draft.meta = testFullTableRegion.subjectMeta;
-        draft.data = testFullTableRegion.results;
+      render(<MapBlockInternal {...testBlockPropsMixed} />);
+
+      await waitFor(() => {
+        expect(
+          screen.getByLabelText('2. Select a location'),
+        ).toBeInTheDocument();
       });
+    });
 
-      render(<MapBlock {...testBlockPropsRegion} />);
+    test('includes all locations and is ungrouped if does not contain local authorities', async () => {
+      render(<MapBlockInternal {...testBlockProps} />);
+
+      await waitFor(() => {
+        const select = screen.getByLabelText(
+          '2. Select a Local Authority District',
+        );
+
+        expect(select).toBeVisible();
+
+        expect(select.children).toHaveLength(4);
+        expect(select.children[0]).toHaveTextContent('None selected');
+        expect(select.children[1]).toHaveTextContent('Leeds');
+        expect(select.children[2]).toHaveTextContent('Manchester');
+        expect(select.children[3]).toHaveTextContent('Sheffield');
+      });
+    });
+
+    test('includes all locations and is grouped if contains local authorities', async () => {
+      const testBlockProps2: MapBlockProps = {
+        ...testMapConfiguration,
+        id: 'testMap',
+        axes: testMixedLocationsAxes as MapBlockProps['axes'],
+        legend: testMapConfiguration.legend as LegendConfiguration,
+        meta: testsMixedLocationsFullTableMeta,
+        data: testsMixedLocationsTableData,
+      };
+
+      render(<MapBlockInternal {...testBlockProps2} />);
 
       await waitFor(() => {
         expect(
           screen.getByLabelText('2. Select a location'),
         ).toBeInTheDocument();
       });
+      const select = screen.getByLabelText('2. Select a location');
+
+      expect(select.children[0]).toHaveTextContent('None selected');
+
+      const groups = within(select).getAllByRole('group');
+      expect(groups).toHaveLength(4);
+
```

Comment:
We should probably do some checks to see if the group labels are correct (and not just the options themselves)